pkg/clusterlogging/clusterlogging.go

```diff
@@ -0,0 +1,190 @@
+package clusterlogging
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	clov1 "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ClusterLoggingBuilder provides struct for clusterLogging object.
+type ClusterLoggingBuilder struct {
+	// ClusterLogging definition. Used to create clusterLogging object with minimum set of required elements.
+	Definition *clov1.ClusterLogging
+	// Created clusterLogging object on the cluster.
+	Object *clov1.ClusterLogging
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+	// errorMsg is processed before clusterLogging object is created.
+	errorMsg string
+}
+
+// NewClusterLoggingBuilder method creates new instance of builder.
+func NewClusterLoggingBuilder(
+	apiClient *clients.Settings, name, nsname string) *ClusterLoggingBuilder {
+	glog.V(100).Infof("Initializing new clusterLogging structure with the following params: name: %s, namespace: %s",
+		name, nsname)
+
+	builder := &ClusterLoggingBuilder{
+		apiClient: apiClient,
+		Definition: &clov1.ClusterLogging{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the clusterLogging is empty")
+
+		builder.errorMsg = "The clusterLogging 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The namespace of the clusterLogging is empty")
+
+		builder.errorMsg = "The clusterLogging 'namespace' cannot be empty"
+	}
+
+	return builder
+}
+
+// Pull returns clusterLogging object if found.
+func (builder *ClusterLoggingBuilder) Pull() (*clov1.ClusterLogging, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting clusterLogging %s in namespace %s",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	clusterLogging := &clov1.ClusterLogging{}
+	err := builder.apiClient.Get(context.Background(), goclient.ObjectKey{
```

Comment:
Please replace it by following logic:

```
if !builder.Exists() {
	return nil, fmt.Errorf("ClusterLogging object %s doesn't exist", name)
}
```

Reply:
Pleas check Pull functions and use it as example.

Reply:
but this is a Pull function. How you whant me to replace Pull with Exists?
It will be a dead-loop, Exists method uses Pull already

Reply:
I renamed it to Get